### ISSUE TITLE
Any way to set it up without cocoa pods?

### ISSUE THREAD

**Issue by renato-bvdash** [heap/react-native-heap#22]
ANY way to set react-native-heap without cocoa pods?
This wouldn't be such an inconvenience if it weren't for the fact that I keep getting errors like:
```
jest-haste-map: Haste module naming collision:
  Duplicate module name: react-native
  Paths: /Users/.../ios/Pods/React/package.json collides with /Users/.../node_modules/react-native/package.json
```
which don't allow me to successfully build in android (and then also getting build error in ios which I'm yet unsure if are related).

I found [the solution](http://genobear.com/2019/01/30/react-native-how-to-fix-duplicate-module-name/) is to add all the libraries of React in the Podfile as well as third parties libraries, but I used 10+ third party libraries, and it's making my life hell tracking them all down.

I would appreciate your help in any way I could make my life simpler.

**Comment by jnatkins** [comment]
Hi @renato-bvdash, we don't currently have a way to set up react-native-heap without Cocoapods, although it is something we plan on doing eventually. In the meantime, the easiest way for us to help you would be if you can share the contents of your Podfile, and we can advise on the best way to fix it up to work with this module.

Thanks!

**Comment by zachrnolan** [comment]
Would also like a way to use this library without Cocoapods.

**Comment by renato-bvdash** [comment]
thanks @jnatkins , but first I would have to know how to set up cocoa pods, because last time I did, it broke my entire app. After guessing all my node modules, I kept having lots of issues with XCode which in short saw two copies of the same node modules.
Some answers in internet suggested deleting the default references to these node modules, but it just further broke the project. In the end I just had to completely roll back.

**Comment by jnatkins** [comment]
We’re working on removing the CocoaPods restriction. Will hopefully be available within the next week or so

**Comment by jnatkins** [comment]
Have another recommendation: you can manually link the native libraries in XCode. You can find some instructions on how to do that here: https://docs.heapanalytics.com/docs/ios#section-manual-ios-installation

**Comment by Maj0rPaine** [comment]
@jnatkins How is the manual installation a work around for RN projects? Are you recommending that we manage tracking without a native module or with our own until an alternative RN installation method is available?

**Comment by GrandPoohBear** [comment]
Hey folks!  I've got a quick update here.  We just merged a change (https://github.com/heap/react-native-heap/pull/62) that will make it pretty easy to set this up without Cocoapods, with only a few manual steps that are in the process of being _nicely_ documented.  This will roll into the next v0.2.0 release.

Essentially (i.e. the non-illustrated abbreviated version), this will involve finding the Xcode project for RNHeap in your node_modules, dragging it into your app's project, then linking the product manually.  We'll break it out step-by-step in case that seems a little daunting.  It's a procedure you're probably very familiar with if you've linked other npm packages manually.  Cocoapods will continue to work, and is still our preferred way of installing the library in your app.

We've abandoned work on getting this to work properly with `react-native link` as there are some changes forthcoming in the React Native CLI that will make the whole thing unnecessary.  I have high hopes that we'll all have a lot less headache incorporating native components in the future!
